### ISSUE TITLE
4.12.0-4.12.1 visibility check regression issue

### ISSUE THREAD

**Issue by maximkoshelenko** [cypress-io/cypress#8190]
### Current behavior:
Starting from 4.12.0 should('not.be.visible') check fails.

### Desired behavior:
During running same test on 4.10.0 - 4.11.0 test pass. 

### Test code to reproduce
```js
    cy.visit('https://live-mdmc.pantheonsite.io');
        cy.get('#global-search-block-form')
      .should('not.be.visible');
    cy.get('#searchAnchor')
      .click();
    cy.get('#global-search-block-form')
      .should('be.visible');
    cy.get('button[aria-label=Close]')
      .click();
    cy.get('#global-search-block-form')
      .should('not.be.visible');
```

### Versions
4.12.0-4.12.1 (Mac, Windows, Linux)

**Comment by jennifer-shehane** [comment]
I can confirm this from the test code provided. This DOM element disappears after clicking the 'close' button in the app - so should no longer be visible, but the test fails saying it is still visible.

![](http://g.recordit.co/PbT6ql3d8Y.gif)

This is a regression introduced in 4.12.0, specifically in https://github.com/cypress-io/cypress/pull/8095 (I compared the commit before and this commit - it fails in this commit).

### Repro

I've narrowed down the reproducible issue to the following. It's still a bit complex, so I'm not exactly sure which piece is causing the issue.

`index.html`

```html
<html>
<head>
  <style>
    .global-search-modal {
      opacity: 0;
      pointer-events: none;
      position: fixed;
      top: 0;
      left: 0;
      width: 100%;
    }

    .open {
      opacity: 1;
      pointer-events: auto;
    }
  </style>
</head>
<body>
  <header style="padding: 30px 0">
    <div style="display: flex;">
      <div style="display: flex; margin-left: auto">
        <nav>
          <ul>
            <li style="width: 500px;"></li>
          </ul>
        </nav>
        <button class="global-search">Search</button>
        <div class="global-search-modal">
          <div style="display: flex; align-items: center; height: 168px;">
            <form id="global-search-block-form" style="position: relative; margin-left: auto;">
              <input />
            </form>
            <button class="close">Close</button>
          </div>
        </div>
      </div>
    </div>
  </header>
  <div style="height: 1000px"></div>
<script>
  const $searchModal = document.querySelector('.global-search-modal');
  var isModalOpen = false;

  document.querySelector('.global-search').addEventListener('click', function () {
    if (isModalOpen) {
      isModalOpen = false;
      $searchModal.classList.remove('open');
    } else {
      isModalOpen = true;
      $searchModal.classList.add('open');
    }
  })

  document.querySelector('.global-search-modal .close').addEventListener('click', function () {
    isModalOpen = false;
    $searchModal.classList.remove('open');
  })
</script>
</body>
</html>
```

`spec.js`

```js
it('test', () => {
  cy.visit('index.html');
  cy.get('#global-search-block-form')
    .should('not.be.visible');
  cy.get('.global-search')
    .click();
  cy.get('#global-search-block-form')
    .should('be.visible');
  cy.get('.close')
    .click();
  cy.get('#global-search-block-form')
    .should('not.be.visible');
})
```
### 4.11.0

<img width="1457" alt="Screen Shot 2020-08-07 at 2 30 03 PM" src="https://user-images.githubusercontent.com/1271364/89623379-82e84c00-d8ba-11ea-8541-9e0a77afc3d6.png">


### 4.12.0

<img width="1469" alt="Screen Shot 2020-08-07 at 2 29 17 PM" src="https://user-images.githubusercontent.com/1271364/89623326-706e1280-d8ba-11ea-853a-c4e7d9c0acc9.png">

**Comment by Bkucera** [comment]
This is due to the element being 'hidden' with `opacity:0` and `pointer-events:none`. In previous versions we incorrectly considered some visible pointer-events:none elements as being hidden, which was fixed in 4.12.0.

So now the Cypress considers the element visible since `opacity:0` does not affect visibility checks currently, making this a duplicate of #4474
